packages/core/src/feature-app-manager.ts

```diff
@@ -60,7 +60,7 @@ export interface FeatureAppDefinition<
   create(env: FeatureAppEnvironment<TFeatureServices, TConfig>): TFeatureApp;
 }
 
-export type ModuleLoader = (url: string) => Promise<unknown>;
+export type ModuleLoader = (url: string, scope?: string) => Promise<unknown>;
```

Comment:
Scope must be a unique identifier for a feature app. It is the global name of the module. Currently the name id taken from the url of the module. Providing a meaningful name requires an API change. The id must b known to the feature app at build time and the integrator at runtime through feature app configuration.

Reply:
scope can be set to a unique name, as long as the onload listener of the script is picking it up and removing it from the window. This is the way SystemJs works.